### PR TITLE
Update create-item.lua

### PR DESCRIPTION
**Pull request by Tjudge1**
Fixed issues with spawning creature based items. Vermin, pets, eggs, fish, raw fish, and remains now spawn and stack correctly.

Fixes: [DFHack/dfhack#4689](https://github.com/DFHack/dfhack/issues/4689)
Fixes: [DFHack/dfhack#4507](https://github.com/DFHack/dfhack/issues/4507)
Fixes: [DFHack/dfhack#3674](https://github.com/DFHack/dfhack/issues/3674)

### CODE REVIEW

**Comment by Tjudge1** [comment]
pre-commit.ci autofix

**Comment by myk002** [comment]
Could you edit the PR description to include lines like:

Fixes: https://github.com/DFHack/dfhack/issues/4689

for the issues that this PR should fix?

File: modtools/create-item.lua [hunk 1]
```diff
@@ -1,5 +1,6 @@
 -- creates an item of a given type and material
 --author expwnent
+--In the year 2024 dikbutdagrate assumed the identity of tjudge1 and edited this file. Now it spawns vermin, pets, eggs, fish, raw fish, and remains correctly.
```
Comment (myk002) [inline modtools/create-item.lua:3]: cute, but this is what the git history is for. if we added a line like this for every edit, we'd have more header comments than code.

File: modtools/create-item.lua [hunk 2]
```diff
@@ -326,16 +336,27 @@ function hackWish(accessors, opts)
         until count
     end
     if not mattype or not itemtype then return end
-    if df.item_type.attrs[itemtype].is_stackable then
+    if not typesThatUseCreaturesExceptCorpses[df.item_type[itemtype]] and df.item_type.attrs[itemtype].is_stackable then
         return createItem({mattype, matindex}, {itemtype, itemsubtype}, quality, unit, description, count)
     end
+    if typesThatUseCreaturesExceptCorpses[df.item_type[itemtype]] and df.item_type.attrs[itemtype].is_stackable then
+        return createItem({matindex, casteId}, {itemtype, itemsubtype}, quality, unit, description, count)
+    end
```
Comment (myk002) [inline modtools/create-item.lua:344]: There is too much duplication here. How about this?
```lua
    if df.item_type.attrs[itemtype].is_stackable then
        local mat = typesThatUseCreaturesExceptCorpses[df.item_type[itemtype]] and {matindex, casteId} or {mattype, matindex}
        return createItem(mat, {itemtype, itemsubtype}, quality, unit, description, count)
    end
```

File: modtools/create-item.lua [hunk 3]
```diff
@@ -326,16 +336,27 @@ function hackWish(accessors, opts)
...
     local items = {}
     for _ = 1,count do
         if itemtype == df.item_type.CORPSEPIECE or itemtype == df.item_type.CORPSE then
             table.insert(items, createCorpsePiece(unit, bodypart, partlayerID, matindex, casteId, corpsepieceGeneric))
         else
-            for _,item in ipairs(createItem({mattype, matindex}, {itemtype, itemsubtype}, quality, unit, description, 1)) do
-                table.insert(items, item)
+            if typesThatUseCreaturesExceptCorpses[df.item_type[itemtype]] then
```
Comment (myk002) [inline modtools/create-item.lua:350]: you can do the same condensing here. Factor out the value of `mat` and avoid duplicating the for loop

**Review by myk002 — COMMENTED** [review]
could you add a line (or lines) to `changelog.txt` explaining (from a player's perspective) what has been fixed?

**Review by Tjudge1 — COMMENTED** [review]
Sure thing.

File: modtools/create-item.lua (hunk 2 above)
Comment (Tjudge1) [reply modtools/create-item.lua:344]: Wait, did you test this?

File: changelog.txt [hunk 4]
```diff
@@ -17,6 +17,10 @@ Template for new versions:
 ## New Features
 
 ## Fixes
+- `gui/create-item`: fix items of type "VERMIN", "PET", "REMANS", "FISH", "RAW FISH", and "EGG" no longer spawn creature item "nothing."
+Items will now spawn correctly, and will be of the creature type and creature caste that selected by the user. Items of these  types will also stack correctly when needed.
+- `modtools/create-item`:  fix items of type "VERMIN", "PET", "REMANS", "FISH", "RAW FISH", and "EGG" no longer spawn creature item "nothing"s.
+Items will now spawn correctly, and will be of the creature type and creature caste that selected by the user. Items of these  types will also stack correctly when needed.
```
Comment (myk002) [inline changelog.txt:23]: this is the "Template for new versions" section -- the actual changelog lines should go in the "Future" version section below

File: changelog.txt [hunk 5]
```diff
@@ -17,6 +17,10 @@ Template for new versions:
 ## New Features
 
 ## Fixes
+- `gui/create-item`: fix items of type "VERMIN", "PET", "REMANS", "FISH", "RAW FISH", and "EGG" no longer spawn creature item "nothing."
```
Comment (myk002) [inline changelog.txt:20]: REMANS -> REMAINS

"no longer spawn creature item "nothing."" -> "being created without a valid creature type"

File: modtools/create-item.lua (hunk 2 above)
Comment (myk002) [reply modtools/create-item.lua:343]: not tested, just written as a suggestion for code improvement. If you have two chunks of code that are semantically and structurally identical except for a small difference, it is usually better to factor that difference out and keep the structure simpler.

**Pull request closed without merging**